### DISCUSSION TITLE
[Feature Request] _Make atom number change possible in training_

### DISCUSSION THREAD

**Discussion by jtyang-chem** [deepmodeling/deepmd-kit#821] [Ideas]
**Summary**

as the tiltle

**Detailed Description**

I've got tons of xyz frames from MD traj, for time saving, in every frame I only draw the atoms in some region then calculate them, so the atom numbers of frames are different, in what way I can   use them for training?

**Comment by njzjz** [comment]
See #486.

**Comment by njzjz** [comment]
@amcadmus @tuoping Shall we add an example and/or document about this?

**Comment by jtyang-chem** [comment]
@njzjz Thx for the quick reply, and sorry I started an old question. I saw the post, because all I got is energies and xyzs, not some standard outputs and may with different atom numbers, which means the systems list length will over a million, and each system will only have one set which contains only one coord and energy, would it cause a trouble? Or do you have any advice?

**Comment by njzjz** [comment]
A million? Are you sure it's 1,000,000? Sorry but I cannot image that thing. You may consider to reduce your training set...

>  each system will only have one set

Also, are you sure there is no same atom numbers in these 1,000,000 structures?

**Comment by jtyang-chem** [comment]
@njzjz On training set, yes, I can reduce them, it's big because I got energies from xtb.
On atom numbers. That's related to the region I choose as my xyz. Yes, there could be some. My system is a solution with two kinds of molecules, though two xyz have same atoms, the molecules in it can be different, also, there could be some xyz with all same types and numbers, still, one can't tell how many of them without a check and I don't want to check them if there is no need to...

**Comment by njzjz** [comment]
You may consider to use [`dpdata.MultiSystems`](https://github.com/deepmodeling/dpdata#load-data), a class to handle such systems, if you can write some Python scripts.

**Comment by jtyang-chem** [comment]
@njzjz  Thanks for your kind reply, I'll try.
